ntlm.go

```diff
@@ -212,11 +309,22 @@ func (auth *ntlmAuth) NextBytes(bytes []byte) ([]byte, error) {
 	var lm, nt []byte
 	if (flags & _NEGOTIATE_EXTENDED_SESSIONSECURITY) != 0 {
 		nonce := clientChallenge()
-		var lm_bytes [24]byte
-		copy(lm_bytes[:8], nonce[:])
-		lm = lm_bytes[:]
-		nt_bytes := ntlmSessionResponse(nonce, challenge, auth.Password)
-		nt = nt_bytes[:]
+		if (flags & _NEGOTIATE_TARGET_INFO) != 0 {
+			timestamp := time.Now()
+			targetInformationBytes, target, err := getTargetInformationData(bytes)
+			if err != nil {
+				return nil, err
+			}
+
+			nt = getNTLMV2Response(target, auth.UserName, auth.Password, challenge, nonce, targetInformationBytes, timestamp)
```

Comment:
I understand you are trying to encapsulate here, but getNTLMV2Response and getLMV2Response takes many of the same inputs, and does many of the same things.

Wouldn't it make sense to have one function that returned both the NTLM and LM response at the same time?

Reply:
You're right, the purpose of the encapsulation can be easily achieved with a couple of comments on the blocks that construct the two payloads.